infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/KZG.java

```diff
@@ -0,0 +1,118 @@
+ */
+
+package tech.pegasys.teku.kzg;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.infrastructure.io.resource.ResourceLoader;
+import tech.pegasys.teku.kzg.impl.KZG4844;
+import tech.pegasys.teku.kzg.impl.KzgException;
+import tech.pegasys.teku.kzg.impl.ckzg.CkzgLoader;
+
+/**
+ * Implements the standard KZG functions needed for the EIP-4844 specification.
+ *
+ * <p>This package strives to implement the KZG standard as used in the Eth2 specification and is
+ * the entry-point for all KZG operations in Teku. Do not rely on any of the classes used by this
+ * one conforming to the specification or standard.
+ */
+public final class KZG {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static KZG4844 kzgImpl;
+
+  static {
+    resetKzgImplementation();
+  }
+
+  public static void setKzgImplementation(final KZG4844 kzgImpl) {
+    KZG.kzgImpl = kzgImpl;
+  }
+
+  public static void resetKzgImplementation() {
+    if (CkzgLoader.INSTANCE.isPresent()) {
+      kzgImpl = CkzgLoader.INSTANCE.get();
+      LOG.info("KZG: loaded CKZG library");
+    } else {
+      throw new KzgException("Failed to load CKZG library.");
+    }
+  }
+
+  public static KZG4844 getKzgImpl() {
+    return kzgImpl;
+  }
+
+  public static void resetTrustedSetup() {
+    try {
+      kzgImpl.resetTrustedSetup();
+    } catch (final KzgException ex) {
+      LOG.trace("Trying to reset KZG trusted setup which is not loaded");
+    }
+  }
+
+  public static void loadTrustedSetup(final String path) {
+    final String filePath;
+    try {
+      filePath = copyResourceToTempFile(path);
+    } catch (final IOException ex) {
+      throw new KzgException(
+          String.format("Failed to copy resource '%s' to temporary file", path), ex);
+    }
+    kzgImpl.loadTrustedSetup(filePath);
+  }
+
+  public static KZGProof computeAggregateKzgProof(final List<Bytes> blobs) {
+    return kzgImpl.computeAggregateKzgProof(blobs);
+  }
+
+  public static boolean verifyAggregateKzgProof(
+      final List<Bytes> blobs, final List<KZGCommitment> kzgCommitments, final KZGProof kzgProof) {
+    return kzgImpl.verifyAggregateKzgProof(blobs, kzgCommitments, kzgProof);
+  }
+
+  public static KZGCommitment blobToKzgCommitment(final Bytes blob) {
+    return kzgImpl.blobToKzgCommitment(blob);
+  }
+
+  public static boolean verifyKzgProof(
+      final KZGCommitment kzgCommitment,
+      final Bytes32 z,
+      final Bytes32 y,
+      final KZGProof kzgProof) {
+    return kzgImpl.verifyKzgProof(kzgCommitment, z, y, kzgProof);
+  }
+
+  private static String copyResourceToTempFile(final String path) throws IOException {
```

Comment:
could we pass-through the file if it is a local path?

Reply:
yes, but I guess we will need absolute path
also how could I recognize it?

Reply:
is you switch path from `String` to `URI` and then `getScheme().equalsIgnoreCase("file")` ?

Reply:
Thank you
also I decided to pass URL there instead of String, so it's clear that it comes with scheme and not just path